src/Codeception/Module/WebDriver.php

```diff
@@ -499,7 +499,20 @@ public function _after(TestInterface $test)
     public function _failed(TestInterface $test, $fail)
     {
         $this->debugWebDriverLogs($test);
-        $filename = preg_replace('~\W~', '.', Descriptor::getTestSignature($test));
+        $example = $test->getMetadata()->getCurrent('example');
```

Comment:
this should not be a fix for WebDriver only. The solution should be more general to include all modules of Codeception

Reply:
I found using addReport function in two modules 'src/Codeception/Lib/InnerBrowser.php' and 'src/Codeception/Module/WebDriver.php'. Are you expecting to add corresponding support for the both? If so I will add. Otherwise if I just remove new code from WebDriver.php module then this commit will be useful, because it will not fix the main issue - damaged HTML report in tests with examples.

Reply:
On the other hand base names of files for all reports are expected to be unified. If so we can add new method to generate base filename considering the name of suite, test and example index. This method will simplify further support on another modules.

Reply:
@davertmik we are waiting for your response here.

Reply:
> Are you expecting to add corresponding support for the both?

Wow, I'm really late here. But yes I expect this to be implemented for both modules. As a method inside of a trait or whatever...